### PR TITLE
gradientOp can take model as inputs, UpdateOp can update from gradients

### PR DESCRIPTION
**Pull request by vbvg2008**

### CODE REVIEW

File: fastestimator/backend/update_model.py [hunk 1]
```diff
@@ -23,7 +23,8 @@
 
 
 def update_model(model: Union[tf.keras.Model, torch.nn.Module],
-                 loss: Union[tf.Tensor, torch.Tensor],
+                 loss: Union[tf.Tensor, torch.Tensor] = None,
+                 gradients: List[Union[tf.Tensor, torch.Tensor]] = None,
```
Comment (TortoiseHam) [inline fastestimator/backend/update_model.py:27]: Optional[List...

File: fastestimator/backend/update_model.py [hunk 2]
```diff
@@ -23,7 +23,8 @@
 
 
 def update_model(model: Union[tf.keras.Model, torch.nn.Module],
-                 loss: Union[tf.Tensor, torch.Tensor],
+                 loss: Union[tf.Tensor, torch.Tensor] = None,
```
Comment (TortoiseHam) [inline fastestimator/backend/update_model.py:26]: Optional[Union...

File: fastestimator/op/tensorop/gradient/gradient.py [hunk 3]
```diff
@@ -30,25 +30,33 @@ class GradientOp(TensorOp):
...
+        model: The model instance to compute gradients with respect to, mutually exclusive with "inputs".
         mode: What mode(s) to execute this Op in. For example, "train", "eval", "test", or "infer". To execute
             regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
             like "!infer" or "!train".
     """
     def __init__(self,
-                 inputs: Union[str, List[str]],
                  finals: Union[str, List[str]],
                  outputs: Union[str, List[str]],
+                 inputs: Union[str, List[str]] = None,
```
Comment (TortoiseHam) [inline fastestimator/op/tensorop/gradient/gradient.py:44]: None, str, ...

File: fastestimator/op/tensorop/gradient/gradient.py [hunk 4]
```diff
@@ -30,25 +30,33 @@ class GradientOp(TensorOp):
...
+                 model: Union[tf.keras.Model, torch.nn.Module] = None,
```
Comment (TortoiseHam) [inline fastestimator/op/tensorop/gradient/gradient.py:45]: None, tf.keras.Model, ...

File: fastestimator/op/tensorop/gradient/gradient.py [hunk 5]
```diff
@@ -30,25 +30,33 @@ class GradientOp(TensorOp):
     """Return the gradients of finals w.r.t. inputs.
 
     Args:
-        inputs: The tensor(s) to compute gradients with respect to.
         finals: The tensor(s) to compute gradients from.
         outputs: The key(s) under which to save the gradients.
+        inputs: The tensor(s) to compute gradients with respect to, mutually exclusive with "model".
```
Comment (TortoiseHam) [inline fastestimator/op/tensorop/gradient/gradient.py:35]: with `model`

File: fastestimator/backend/update_model.py (hunk 2 above)
Comment (TortoiseHam) [inline fastestimator/backend/update_model.py]: Union[None, ...

**Review by TortoiseHam — APPROVED** [review]
LG
